### ISSUE TITLE
Sources at a given position

### ISSUE THREAD

**Issue by tischi** [bigdataviewer/bigdataviewer-playground#160]
@NicoKiaru 

Given all the advancements with only rendering sources that are visible in the current field of view, are there now maybe methods that would allow us to improve this?

```java
    /**
     * Is the point pt located inside the source  at a particular timepoint ?
     * Looks at highest resolution whether the alpha value of the displayed pixel is zero
     * TODO TO think Alternative : looks whether R, G and B values equal zero - source not present
     * Another option : if the display RGB value is zero, then consider it's not displayed and thus not selected
     * - Convenient way to adjust whether a source should be selected or not ?
     * TODO : Time out if too long to access the data
     * @param sac
     * @param pt
     * @return
     */
    public static boolean isSourcePresentAt(SourceAndConverter sac, int timePoint, RealPoint pt) {
```

Something that does not rely on the `Converter` would be good I think, because we cannot (should not) control which converters people may use. Is there something we can do (and that would also work with the warped sources)?

**Comment by NicoKiaru** [comment]
Difficult!

The advancement in the rendering only works nicely with non-warped source. If we want to make this works with warped source, then I'm not source we can do better than relying on converters.

Now I think this particular method is not really used anymore. The editor mode accessible with the E key is amuch better way to select a subset of sources

**Comment by tischi** [comment]
> The advancement in the rendering only works nicely with non-warped source. If we want to make this works with warped source, then I'm not source we can do better than relying on converters.

1. Could we make a `if ( source instanceof WarpedSource )` distinction and only then use the converter, but otherwise the other approach
2. Where could I find code for the other (rendering based) approach?

> The editor mode accessible with the E key is a much better way to select a subset of sources

I would need programmatic access a method `getSourcesAtPosition( xyzt )`. Is there such a method available or could you point me to code that would enable me to write one?

**Comment by NicoKiaru** [comment]
>     1. Could we make a `if ( source instanceof WarpedSource )` distinction and only then use the converter, but otherwise the other approach

Hard. Because if it is a TransformedSource, it can be a TransformedSource from a WarpedSource or a TransformedSource of a SpimSource, or a TransformedSource of a TransformedSource...

> Where could I find code for the other (rendering based) approach?

Do you mean the fact that Sources outside of the screen are ignored ? I think it ressembles what we discussed when we wanted to modify the Projector.

> I would need programmatic access a method getSourcesAtPosition( xyzt ). Is there such a method available or could you point me to code that would enable me to write one?

Ah. I don't know if such method exists for now.

**Comment by tischi** [comment]
> Ah. I don't know if such method exists for now.

But for your bdv-selector, don't you have something like this?

Otherwise I'll make a PR adding a method that I am typically using, where I check whether the calibrated position, if converted to the source's voxel units is within the source's interval.

**Comment by tischi** [comment]
...ok, I had a look in the [bdv-selector](https://github.com/NicoKiaru/bigdataviewer-selector/blob/4d6b615a24b93cbffe3510af3227e04fd0fdd217/src/main/java/ch/epfl/biop/bdv/select/SourceSelectorOverlay.java#L137), I think something like this would work, but maybe it's an overkill for just checking whether a single point intersects with a source. As mentioned above, I'll make a PR with some simpler method.

**Comment by NicoKiaru** [comment]
Yeah, it's complicated. I've just done something in bigdataviewer-playground-display with 'AlphaSource' which can be used to know if any source is present or not at a certain point, but that's some extra difficulties, and probably out of scope for this repo: https://github.com/BIOP/bigdataviewer-playground-display/tree/master/src/main/java/bdv/util/source/alpha

**Comment by NicoKiaru** [comment]
See https://github.com/bigdataviewer/bigdataviewer-playground/issues/188 for a related discussion

**Comment by NicoKiaru** [comment]
Closing for now. Please re-open if needed
